airflow/executors/fargate_executor.py

```diff
@@ -0,0 +1,327 @@
+        else:
+            self.executor_config_function: ExecutorConfigFunctionType = default_task_id_to_fargate_options_function()
+        self.region = configuration.conf.get('fargate', 'region')
+        self.cluster = configuration.conf.get('fargate', 'cluster')
+        self.active_workers: Optional[FargateTaskCollection] = None
+        self.ecs = None
+
+    def start(self):
+        """Initialize Boto3 ECS Client, and other internal variables"""
+        self.active_workers = FargateTaskCollection()
+        # NOTE: See https://boto3.amazonaws.com/v1/documentation/api/latest/guide/configuration.html for authentication
+        # and access-key management. You can store an environmental variable, setup aws config from console, or use IAM
+        # roles.
+        self.ecs = boto3.client('ecs', region_name=self.region)
+
+    def sync(self):
+        """Updates the status of all Airflow Tasks currently pending"""
+        # Check on all active workers
+        all_task_arns = self.active_workers.get_all_arns()
+        if not all_task_arns:
+            self.log.debug("No active tasks, skipping sync")
+            return
+        self.log.debug("Active Workers: {}".format(all_task_arns))
+        describe_dict = self.ecs.describe_tasks(tasks=all_task_arns, cluster=self.cluster)
+        self.log.debug('ECS Response: %s' % describe_dict)
+        # check & handle the failures first in the JSON response.
+        if describe_dict['failures']:
+            self._sync_api_failures(describe_dict['failures'])
+        # check & handle airflow sucesses & failures next
+        updated_tasks = describe_dict['tasks']
+        for task_json in updated_tasks:
+            task = FargateTask(task_json)
+            # Update the active workers with new statuses. Consequtive task failure counter resets back to 0.
+            self.active_workers.update_task(task)
+            # get state of current task
+            task_state = task.get_task_state()
+            task_key = self.active_workers.arn_to_key[task.task_arn]
+            # mark finished tasks as either a success/failure
+            if task_state == State.FAILED:
+                self.fail(task_key)
+            elif task_state == State.SUCCESS:
+                self.success(task_key)
+            # if task is no longer running then remove from active workers and place in finished
+            if task_state != State.RUNNING:
+                self.log.debug('Task %s marked as %s after running on %s' % (task_key, task_state, task.task_arn))
+                self.active_workers.pop_by_key(task_key)
+
+    def _sync_api_failures(self, arn_failures):
+        """
+        In the case of an API failure this executor will try a set amount of retries. Sometimes
+        one call to AWS might fail intermittently, and just needs to be called again in the next heartbeat. If
+        a given API call fails consecutively to a 'MAX_FAILURE_CHECKS' number of calls, the task is marked as failed.
```

Comment:
Please use `:param` and `:type` in the doc strings. This way the sphinx build can generate a clearer documentation. :)